daemon/graphdriver/overlay2/check.go

```diff
@@ -8,17 +8,20 @@ import (
 	"os"
 	"path"
 	"path/filepath"
+	"strings"
+	"syscall"
 
 	"github.com/Sirupsen/logrus"
 	"github.com/docker/docker/pkg/system"
 	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
-// hasOpaqueCopyUpBug checks whether the filesystem has a bug
+// hasOpaqueCopyUpBugOrEnableRedirectDir checks whether the filesystem has a bug
 // which copies up the opaque flag when copying up an opaque
-// directory. When this bug exists naive diff should be used.
-func hasOpaqueCopyUpBug(d string) error {
+// directory or the kernel enable CONFIG_OVERLAY_FS_REDIRECT_DIR.
+// When these exist naive diff should be used.
+func hasOpaqueCopyUpBugOrEnableRedirectDir(d string) error {
```

Comment:
Feels too long a name. May be new function should be in a separate helper function of its own instead of single function checking both.